src/txn/txn.c

```diff
@@ -926,27 +901,33 @@ __txn_timestamp_usage_check(WT_SESSION_IMPL *session, WT_TXN_OP *op, WT_UPDATE *
 }
 
 /*
- * __txn_fixup_prepared_update --
- *     Fix the history store record with the max stop time point if we commit the prepared update.
+ * __txn_fixup_hs_update --
+ *     Fix the history store update with the max stop time point if we commit the prepared update.
  */
 static int
-__txn_fixup_prepared_update(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, WT_UPDATE *fix_upd)
+__txn_fixup_hs_update(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor)
 {
+    WT_DECL_ITEM(hs_value);
     WT_DECL_RET;
-    WT_ITEM hs_value;
-    WT_TIME_WINDOW tw;
+    WT_TIME_WINDOW *hs_tw, tw;
     WT_TXN *txn;
-    uint32_t txn_flags;
-#ifdef HAVE_DIAGNOSTIC
-    uint64_t hs_upd_type;
     wt_timestamp_t hs_durable_ts, hs_stop_durable_ts;
```

Comment:
We can get rid of these?

Reply:
Why? They are used later.

Reply:
Thanks for clarifying that we cannot pass NULL in the `get_value()` function. Resolving this comment.